### PR TITLE
bugfix: large streaming Markdown rendering delay

### PR DESCRIPTION
**Pull request by harshil-2096**
## Summary

Closes #

## Changes

-

## Checklist

- [ ] `pnpm build`, `pnpm test`, `pnpm typecheck`, `pnpm lint:ci`, and `pnpm format:check` pass locally
- [ ] Tests added/updated where it makes sense
- [ ] No hand-edits to generated code (`packages/sdk`, `.github/fern/openapi/openapi.json`)
- [ ] Docs / `.env.example` updated if configuration or behavior changed

---

> [!NOTE]
> **Medium Risk**
> Changes core streaming/reveal behavior for assistant messages and markdown rendering paths; well-covered by new tests but could affect perceived typing animation or scroll behavior in edge cases.
> 
> **Overview**
> Fixes **large streaming Markdown** feeling stuck by pacing updates and avoiding expensive work on every SSE tick.
> 
> **Assistant text reveal:** When the typewriter-style `useSmooth` reveal falls more than ~4KB behind the raw stream, the container **latches paced mode** and shows throttled snapshots (~10 updates/sec) via `useThrottledMarkdownText`, then flushes the exact text when the stream completes. `useSmooth` is not turned back on for that part (avoids visible rewind). Markdown is only re-rendered when committed display text actually changes.
> 
> **Code fences while streaming:** Only the **still-open, growing** fence body above ~8KB is rendered as plain text instead of Prism; closed large fences stay highlighted. React markdown `components` stay stable via a ref so finished syntax blocks do not remount as more text arrives. `SyntaxHighlighter` is memoized.
> 
> **Thread viewport:** Removes `scroll-smooth` on the message viewport to reduce scroll bounce with assistant-ui auto-scroll on big streams.
> 
> Also bumps `@truefoundry/assistant-ui-runtime` to **0.1.15** across the monorepo and docs/lockfiles.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit dba54b84000343cbd3e36f249f622c5d2b183e6f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/trueforge-ui-sdk/src/atoms/Markdown.tsx [hunk 1]
```diff
@@ -86,11 +114,34 @@ function makeComponents(opts: {
         );
       }
 
+      // Only the unmatched growing fence skips Prism; closed fences stay highlighted.
+      const activeFenceCode = getActiveFenceCode();
+      const isActiveOversizedFence =
+        isStreaming === true &&
+        activeFenceCode != null &&
+        code === activeFenceCode &&
+        code.length > LARGE_STREAMING_FENCE_CHARS;
```
[the remark this answers is not in the document]
Comment (harshil-2096) [reply packages/trueforge-ui-sdk/src/atoms/Markdown.tsx:123]: rare edge case

**Review by govindavashishtha — APPROVED** [review]
